lib/foreman_maintain.rb

```diff
@@ -68,9 +82,29 @@ def init_logger
       # argument to Logger.new will not work as expected
       filename = File.expand_path("#{config.log_dir}/foreman-maintain.log")
       @logger = Logger.new(filename, 10, 10_240_000).tap do |logger|
-        logger.level = config.log_level || Logger::DEBUG
+        logger.level = LOGGER_LEVEL_MAPPING[config.log_level] || Logger::DEBUG
         logger.datetime_format = '%Y-%m-%d %H:%M:%S%z '
       end
+      pickup_log_messages
+    end
+
+    def pickup_log_messages
+      return if @pre_setup_log_messages.empty?
+      @pre_setup_log_messages.each { |msg| logger.info msg }
+      @pre_setup_log_messages.clear
+    end
+
+    def load_custom_configs(file_path)
+      file_path ||= ''
+      custom_configs = {}
+      if File.exist?(file_path)
+        custom_configs = YAML.load(File.open(file_path)) || {}
+      else
+        @pre_setup_log_messages << "Config file #{file_path} not found, using default configuration"
+      end
+      custom_configs
+    rescue => e
+      $stderr.puts "Couldn't load configuration file: #{e.message}"
```

Comment:
We probably should raise an error here: if there is an error in the config file, we should not even continue